File: android-wrapper/app/src/main/java/org/m0skit0/android/hms/unity/ads/RewardAdStatusListenerWrapper.java
package org.m0skit0.android.hms.unity.ads;

import com.huawei.hms.ads.reward.Reward;

public class RewardAdStatusListenerWrapper extends com.huawei.hms.ads.reward.RewardAdStatusListener {

    private final RewardAdStatusListener listener;

    public RewardAdStatusListenerWrapper(final RewardAdStatusListener listener) {
        super();
        this.listener = listener;
    }

    @Override
    public void onRewardAdClosed() {
        super.onRewardAdClosed();
        listener.onRewardAdClosed();
    }

    @Override
    public void onRewardAdFailedToShow(int errorCode) {
        super.onRewardAdFailedToShow(errorCode);
        listener.onRewardAdFailedToShow(errorCode);
    }

    @Override
    public void onRewardAdOpened() {
        super.onRewardAdOpened();
        listener.onRewardAdOpened();
    }

    @Override
    public void onRewarded(Reward reward) {
        super.onRewarded(reward);
        listener.onRewarded(reward);
    }
}

File: android-wrapper/app/src/main/java/org/m0skit0/android/hms/unity/ads/RewardAdStatusListenerWrapper.kt
package org.m0skit0.android.hms.unity.ads

import com.huawei.hms.ads.reward.Reward

class RewardAdStatusListenerWrapper(private val listener: RewardAdStatusListener) : com.huawei.hms.ads.reward.RewardAdStatusListener() {

    override fun onRewardAdClosed() {
        listener.onRewardAdClosed()
    }

    override fun onRewardAdFailedToShow(errorCode: Int) {
        listener.onRewardAdFailedToShow(errorCode)
    }

    override fun onRewardAdOpened() {
        listener.onRewardAdOpened()
    }

    override fun onRewarded(reward: Reward) {
        listener.onRewarded(reward)
    }
}